subsystems/provisioning/provisioning.go

```diff
@@ -359,3 +371,30 @@ func (w *Provisioning) updateHotspotSSID(cfg *Config) {
 		cfg.hotspotSSID = cfg.hotspotSSID[:32]
 	}
 }
+
+func (w *Provisioning) writeWifiPowerSave() error {
+	contents := wifiPowerSaveContentsDefault
+
+	if w.cfg.DisableWifiPowerSave != nil && *w.cfg.DisableWifiPowerSave {
+		contents = wifiPowerSaveContentsDisable
+	}
+
+	if w.cfg.DisableWifiPowerSave != nil && !*w.cfg.DisableWifiPowerSave {
+		contents = wifiPowerSaveContentsEnable
+	}
+
+	isNew, err := agent.WriteFileIfNew(wifiPowerSaveFilepath, []byte(contents))
+	if err != nil {
+		return errw.Wrap(err, "error writing wifi-powersave.conf")
+	}
+
+	if isNew {
+		w.logger.Infof("Updated %s to: %q", wifiPowerSaveFilepath, contents)
+		// Reload NetworkManager to apply changes
+		if err := w.nm.Reload(0); err != nil {
```

Comment:
I confirmed looking at network manager logs that the config file logs without warning or errors